onnxruntime/core/mlas/lib/convolve.cpp

```diff
@@ -1210,7 +1200,11 @@ Return Value:
             TargetThreadCount = MLAS_MAXIMUM_THREAD_COUNT;
         }
 
-        int32_t MaximumThreadCount = MlasPlatform.GetMaximumThreadCount();
+        // Take the thread count either limited by the available pool or the platform limit
+        int32_t MaximumThreadCount = ThreadPoolLimit;
+        if (ThreadPoolLimit <= 0) {
+            MlasPlatform.GetMaximumThreadCount();
```

Comment:
You're calling GetMaximumThreadCount but throwing away the result?

Reply:
Fixed.